spec/ddtrace/contrib/rails/rails_active_job_spec.rb

```diff
@@ -2,77 +2,123 @@
 # This module tests the right integration between Sidekiq and
 # Rails. Functionality tests for Rails and Sidekiq must go
 # in their testing modules.
-require 'sidekiq/testing'
-require 'ddtrace/contrib/sidekiq/server_tracer'
+begin
+  require 'sidekiq/testing'
+  require 'ddtrace/contrib/sidekiq/server_tracer'
+rescue LoadError
+  puts 'Sidekiq testing harness not loaded'
+end
+
 begin
   require 'active_job'
 rescue LoadError
   puts 'ActiveJob not supported in this version of Rails'
 end
 
 require 'ddtrace/contrib/rails/rails_helper'
+require 'ddtrace/contrib/active_job/integration'
 
-RSpec.describe 'Rails with Sidekiq' do
+RSpec.describe 'ActiveJob' do
   before { skip unless defined? ::ActiveJob }
-
   include_context 'Rails test application'
 
-  before do
-    allow(ENV).to receive(:[]).and_call_original
-    allow(ENV).to receive(:[]).with('USE_SIDEKIQ').and_return('true')
-  end
+  context 'with active_job instrumentation' do
+    before do
+      Datadog.configure do |c|
+        c.use :active_job
+      end
 
-  before do
-    Sidekiq.configure_client do |config|
-      config.redis = { url: ENV['REDIS_URL'] }
+      # initialize the application
+      app
     end
 
-    Sidekiq.configure_server do |config|
-      config.redis = { url: ENV['REDIS_URL'] }
-    end
+    after do
+      Datadog.registry[:active_job].reset_configuration!
+      Datadog.configuration.reset!
 
-    Sidekiq::Testing.inline!
-  end
-
-  before { app }
+      # unsubscribe the ActiveSupport notifications
+      ActiveSupport::Notifications.unsubscribe('perform.active_job')
```

Comment:
This is a smart way to reset the test suite for this integration, kudos! 🙇

Reply:
😢  ...unfortunately I learned that the [Patcher will only patch once](https://github.com/DataDog/dd-trace-rb/blob/f7431ce273cea753a8bee4f6e64ed2b2b098b04b/lib/ddtrace/contrib/patcher.rb#L23-L36), which means that this unsubscribe prevents the integration from being properly subscribed in subsequent tests. I had to remove the unsubscribe.

It seems like having active_job enabled only impacts the Sidekiq tests, but it just means targeting the appropriate span:

```
# in tests
sidekiq_span = spans.find { |s| s.name == "sidekiq.job" }
active_job_span = spans.find { |s| s.name == "active_job.perform" }
```

Reply:
This snippet looks reasonable ☝️.
We only patch gems once because when instrumentations perform destructive operations (like `#include` or `#prepend`), those can't be undone safely. It's better to work with the language constraints in that case.